File: SCTE65/src/java/other/org/himalay/si/app_desc_baseFactory.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.si;

import java.io.ByteArrayInputStream;
import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.io.PrintStream;
import java.util.ArrayList;
import org.himalay.msgs.runtime.Factory;
import org.himalay.msgs.runtime.FactoryProduct;
import org.himalay.msgs.runtime.Created;
import org.himalay.msgs.runtime.*;

@Created(date = "Sun Sep 14 22:48:37 EDT 2014")
public class app_desc_baseFactory implements Factory {
	public static final int unbound_application_descriptor = 0x67;
	public static final int abstract_service_descriptor = 0x66;
	public static final int privileged_certificate_descriptor = 0x68;
	public static final int application_storage_descriptor = 0x69;
	public static final int application_name_descriptor = 0x01;
	public static final int application_descriptor = 0x00;
	public static final int transport_protocol_descriptor = 0x02;
	public static final int dvbJapp_descriptor = 0x03;
	public static final int dvbJapp_locator_descriptor = 0x04;

	public static boolean isDebug = true;
	public static app_desc_base lastParsedObject = null;
	public static app_desc_base createMsg(DataInputStream istream,
			IntegerHolder iHolder1) throws IOException {
		int iRead = 0;
		descriptor_header desc_header;
		{
			int retVal = 0;
			desc_header = new descriptor_header();
			retVal += desc_header.read(istream);
			iRead = retVal;
		}
		app_desc_base retVal = createMsg(desc_header, istream, iHolder1);
		iHolder1.setValue(iHolder1.getValue() + iRead);
		return retVal;
	}

	public static app_desc_base createMsg(
			org.himalay.si.descriptor_header temp, DataInputStream dis,
			IntegerHolder iHolder) throws IOException {
		int iRead = 0;
		app_desc_base retVal = null;
		try {
			switch (temp.getMessageType()) {
				case 0x67 : {
					retVal = new unbound_application_descriptor();
					retVal.setHeader(temp);
					// has no spread
					iRead += retVal.readNoHeader(dis);
					break;
				}
				case 0x66 : {
					retVal = new abstract_service_descriptor();
					retVal.setHeader(temp);
					{
						// int iLength= retVal.getHeader().length + (0);
						// byte[] ba= new byte[iLength];
						// dis.readFully(ba);
						// dis= new DataInputStream(new
						// ByteArrayInputStream(ba));
					}
					iRead += retVal.readNoHeader(dis);
					break;
				}
				case 0x68 : {
					retVal = new privileged_certificate_descriptor();
					retVal.setHeader(temp);
					// has no spread
					iRead += retVal.readNoHeader(dis);
					break;
				}
				case 0x69 : {
					retVal = new application_storage_descriptor();
					retVal.setHeader(temp);
					{
						// int iLength= retVal.getHeader().length + (0);
						// byte[] ba= new byte[iLength];
						// dis.readFully(ba);
						// dis= new DataInputStream(new
						// ByteArrayInputStream(ba));
					}
					iRead += retVal.readNoHeader(dis);
					break;
				}
				case 0x01 : {
					retVal = new application_name_descriptor();
					retVal.setHeader(temp);
					{
						// int iLength= retVal.getHeader().length + (0);
						// byte[] ba= new byte[iLength];
						// dis.readFully(ba);
						// dis= new DataInputStream(new
						// ByteArrayInputStream(ba));
					}
					iRead += retVal.readNoHeader(dis);
					break;
				}
				case 0x00 : {
					retVal = new application_descriptor();
					retVal.setHeader(temp);
					{
						// int iLength= retVal.getHeader().length + (0);
						// byte[] ba= new byte[iLength];
						// dis.readFully(ba);
						// dis= new DataInputStream(new
						// ByteArrayInputStream(ba));
					}
					iRead += retVal.readNoHeader(dis);
					break;
				}
				case 0x02 : {
					retVal = new transport_protocol_descriptor();
					retVal.setHeader(temp);
					{
						// int iLength= retVal.getHeader().length + (0);
						// byte[] ba= new byte[iLength];
						// dis.readFully(ba);
						// dis= new DataInputStream(new
						// ByteArrayInputStream(ba));
					}
					iRead += retVal.readNoHeader(dis);
					break;
				}
				case 0x03 : {
					retVal = new dvbJapp_descriptor();
					retVal.setHeader(temp);
					{
						// int iLength= retVal.getHeader().length + (0);
						// byte[] ba= new byte[iLength];
						// dis.readFully(ba);
						// dis= new DataInputStream(new
						// ByteArrayInputStream(ba));
					}
					iRead += retVal.readNoHeader(dis);
					break;
				}
				case 0x04 : {
					retVal = new dvbJapp_locator_descriptor();
					retVal.setHeader(temp);
					{
						// int iLength= retVal.getHeader().length + (0);
						// byte[] ba= new byte[iLength];
						// dis.readFully(ba);
						// dis= new DataInputStream(new
						// ByteArrayInputStream(ba));
					}
					iRead += retVal.readNoHeader(dis);
					break;
				}
				default : {
					retVal = new generic_app_desc();
					retVal.setHeader(temp);
					{
						// int iLength= retVal.getHeader().length + (0);
						// byte[] ba= new byte[iLength];
						// dis.readFully(ba);
						// dis= new DataInputStream(new
						// ByteArrayInputStream(ba));
					}
					iRead += retVal.readNoHeader(dis);
					break;
				}
			}
		} catch (IOException ioex) {
			if (isDebug) {
				lastParsedObject = retVal;
			}
			throw ioex;
		}
		iHolder.setValue(iRead);
		return retVal;
	}

	public static abstract class app_desc_base extends BinMessage
			implements
				FactoryProduct {
		public app_desc_base() {
			super();
			// TODO Auto-generated constructor stub
		}
		public abstract void setHeader(org.himalay.si.descriptor_header header);
		public abstract org.himalay.si.descriptor_header getHeader();

	}
}

// End of code

File: libsi/src/bp/app_desc_baseFactory.cpp
//@Generated(date = "Mon Feb 01 22:29:56 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
#include "app_desc_baseFactory.h"
#include "unbound_application_descriptor.h"
#include "abstract_service_descriptor.h"
#include "privileged_certificate_descriptor.h"
#include "application_storage_descriptor.h"
#include "application_name_descriptor.h"
#include "application_descriptor.h"
#include "transport_protocol_descriptor.h"
#include "dvbJapp_descriptor.h"
#include "dvbJapp_locator_descriptor.h"
#include "generic_app_desc.h"
// Namespace
namespace org {
namespace himalay {
namespace si {

using namespace ::org::himalay::msgs::runtime;


org::himalay::si::app_desc_baseFactory::app_desc_base::app_desc_base() {};
org::himalay::si::app_desc_baseFactory::app_desc_base::~app_desc_base() {};
bool app_desc_baseFactory::isDebug = true;
/**
 *
 *app_desc_baseFactory::app_desc_basePtr app_desc_baseFactory::lastParsedObject= NULL;
 *
 */
org_himalay_si_app_desc_baseFactory_app_desc_base__PTR_TYPE app_desc_baseFactory::createMsg(DataInputStream& istream, int& iHolder) throw(IOException)
{
    int retVal= 0;
    //org_himalay_si_descriptor_header__PTR_TYPE temp;
    org_himalay_si_descriptor_header__PTR_TYPE desc_header ;
    org_himalay_si_descriptor_header__NEW( desc_header); ;
    //org_himalay_si_descriptor_header__NEW(temp) ;//new org.himalay.msgs.cpp.Complex@af215c18());
    retVal += desc_header->read(istream);


    org_himalay_si_app_desc_baseFactory_app_desc_base__PTR_TYPE retObj= createMsg( desc_header,  istream,  iHolder);
    iHolder=iHolder+retVal;
    return retObj;
}

org_himalay_si_app_desc_baseFactory_app_desc_base__PTR_TYPE app_desc_baseFactory::createMsg( org_himalay_si_descriptor_header__PTR_TYPE temp, DataInputStream& istream, int& iHolder) throw(IOException)
{
    int retVal= 0;
    org_himalay_si_app_desc_baseFactory_app_desc_base__PTR_TYPE retObj;
    try {
        switch( temp->getMessageType() )        {
        case 0x67 :
        {
            retObj= org_himalay_si_app_desc_baseFactory_app_desc_base__PTR_TYPE((app_desc_baseFactory::app_desc_base *) new unbound_application_descriptor ());
            retObj->setHeader(temp);
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        case 0x66 :
        {
            retObj= org_himalay_si_app_desc_baseFactory_app_desc_base__PTR_TYPE((app_desc_baseFactory::app_desc_base *) new abstract_service_descriptor ());
            retObj->setHeader(temp);
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        case 0x68 :
        {
            retObj= org_himalay_si_app_desc_baseFactory_app_desc_base__PTR_TYPE((app_desc_baseFactory::app_desc_base *) new privileged_certificate_descriptor ());
            retObj->setHeader(temp);
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        case 0x69 :
        {
            retObj= org_himalay_si_app_desc_baseFactory_app_desc_base__PTR_TYPE((app_desc_baseFactory::app_desc_base *) new application_storage_descriptor ());
            retObj->setHeader(temp);
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        case 0x01 :
        {
            retObj= org_himalay_si_app_desc_baseFactory_app_desc_base__PTR_TYPE((app_desc_baseFactory::app_desc_base *) new application_name_descriptor ());
            retObj->setHeader(temp);
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        case 0x00 :
        {
            retObj= org_himalay_si_app_desc_baseFactory_app_desc_base__PTR_TYPE((app_desc_baseFactory::app_desc_base *) new application_descriptor ());
            retObj->setHeader(temp);
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        case 0x02 :
        {
            retObj= org_himalay_si_app_desc_baseFactory_app_desc_base__PTR_TYPE((app_desc_baseFactory::app_desc_base *) new transport_protocol_descriptor ());
            retObj->setHeader(temp);
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        case 0x03 :
        {
            retObj= org_himalay_si_app_desc_baseFactory_app_desc_base__PTR_TYPE((app_desc_baseFactory::app_desc_base *) new dvbJapp_descriptor ());
            retObj->setHeader(temp);
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        case 0x04 :
        {
            retObj= org_himalay_si_app_desc_baseFactory_app_desc_base__PTR_TYPE((app_desc_baseFactory::app_desc_base *) new dvbJapp_locator_descriptor ());
            retObj->setHeader(temp);
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        default :
        {
            retObj= org_himalay_si_app_desc_baseFactory_app_desc_base__PTR_TYPE( (app_desc_baseFactory::app_desc_base *) new generic_app_desc ());
            retObj->setHeader(temp);
            retVal+=retObj->readNoHeader(istream);
            break;
        }
        }
    } catch(IOException& ioex)
    {
        if (isDebug)
        {
//        lastParsedObject = retVal;
        }
        throw ioex;
    }
    iHolder=retVal;
    return retObj;
}
void intrusive_ptr_add_ref(org::himalay::si::app_desc_baseFactory::app_desc_base* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::si::app_desc_baseFactory::app_desc_base* p) {
    if (p->decreaseRef() == 0u) delete p;
};

}
}
} // End Namespace org.himalay.si




// End of code